core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -67,17 +67,16 @@ private[spark] class PythonRDD(
       envVars += ("SPARK_REUSE_WORKER" -> "1")
     }
     val worker: Socket = env.createPythonWorker(pythonExec, envVars.toMap)
+    // Whether is the worker released into idle pool
+    var released = false
 
     // Start a thread to feed the process input from our parent's iterator
     val writerThread = new WriterThread(env, worker, split, context)
 
-    var complete_cleanly = false
     context.addTaskCompletionListener { context =>
       writerThread.shutdownOnTaskCompletion()
       writerThread.join()
-      if (reuse_worker && complete_cleanly) {
-        env.releasePythonWorker(pythonExec, envVars.toMap, worker)
-      } else {
+      if (!released) {
```

Comment:
I think the old version of this logic was slightly easier to understand since the variable names made it clear that we _always_ close the worker if worker re-use is disabled and that we _might_ close the worker if it didn't exit cleanly.

To make this more clear in the updated code, I propose that we modify this conditional to say `if (!reuse_worker || !released)` and maybe add a comment (or change `released` back to `compete_cleanly`).